### PR TITLE
Fix/ssr css variables

### PR DESCRIPTION
**Pull request by m0ksem**
closes https://github.com/epicmaxco/vuestic-ui/issues/3046

Added a build plugin which converts `v-bind(variable)` in <style> to `:style=""`.

Basically, creates a css variable like `--va-[index]-[kebab-name]` and pass it to style using v-bind and replaces vue v-bind in css with generated css variable. This way we ignore vue `useCSSVars` which is not available in cjs build. We also ensure that passed value to css variable is going to be String, which makes it work in Nuxt.

### CODE REVIEW

File: packages/ui/build/plugins/component-v-bind-fix.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    return code.replace(/style="([^"]*)"/, `:style="\`$1;${cssVariablesString}\`"`)
+  }
+
+  if (/\/>$/.test(rootNode)) {
+    const newRootNode = rootNode.replace('/>', ` :style="\`${cssVariablesString}\`"/>`)
+    return code.replace(rootNode, newRootNode)
+  }
+
+  const newRootNode = rootNode.replace('>', ` :style="\`${cssVariablesString}\`">`)
```
[the remark this answers is not in the document]
Comment (m0ksem) [reply packages/ui/build/plugins/component-v-bind-fix.ts:76]: That's the idea. To replace first occurred `>`. Am I doing something wrong here?

File: packages/ui/src/composables/useCSSVariables.ts [hunk 2]
```diff
@@ -2,5 +2,8 @@ import { computed } from 'vue'
 import kebab from 'lodash/kebabCase.js'
 
 export const useCSSVariables = (prefix: string, cb: () => Record<string, string>) => {
-  return computed(() => Object.entries(cb()).map(([key, value]) => ({ [`--${prefix}-${kebab(key)}`]: value })))
+  return computed(() => Object.entries(cb()).reduce((acc, [key, value]) => {
+    acc[`--${prefix}-${kebab(key)}`] = value
```
Comment (aluarius) [inline packages/ui/src/composables/useCSSVariables.ts:6]: Would be nice to have here return value typing as in `useBem`. But up to you.

File: packages/ui/build/plugins/component-v-bind-fix.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
+import { Plugin } from 'vite'
+import kebabCase from 'lodash/kebabCase'
+
+const parseVBinds = (style: string) => {
+  return style
+    .match(/v-bind\((.*)\)/g)
```
Comment (aluarius) [inline packages/ui/build/plugins/component-v-bind-fix.ts:6]: Comment these two lines, unreadable :c

File: packages/ui/build/plugins/component-v-bind-fix.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+const renderCssVariablesString = (vBinds: string[]) => {
+  return vBinds.map((vBind, index) => {
+    return `--va-${index}-${kebabCase(vBind)}: \${String(${vBind})}`
+  }, '').join(';')
+}
+
+const renderObjectGuard = (styleContent: string, binds: string[]) => {
+  const cssVariablesString = renderCssVariablesString(binds)
+  const cssVariablesObject = binds.map((vBind, index) => {
```
Comment (aluarius) [inline packages/ui/build/plugins/component-v-bind-fix.ts:24]: Result will be string, no?

File: packages/ui/build/plugins/component-v-bind-fix.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (aluarius) [reply packages/ui/build/plugins/component-v-bind-fix.ts:76]: Mb you should test it here as at 71 line?

**Review by aluarius — APPROVED** [review]
Logic is pretty hard for understanding, please comment it as much as possible.

File: packages/ui/build/plugins/component-v-bind-fix.ts (hunk 4 above)
Comment (m0ksem) [reply packages/ui/build/plugins/component-v-bind-fix.ts]: It is as string that contains the rendered Object. Renamed.

File: packages/ui/src/composables/useCSSVariables.ts (hunk 2 above)
Comment (m0ksem) [reply packages/ui/src/composables/useCSSVariables.ts:6]: To much effort for it...

File: packages/ui/build/plugins/component-v-bind-fix.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (m0ksem) [reply packages/ui/build/plugins/component-v-bind-fix.ts]: I think it is false positive here, because rootNode is `<tagName attrs>`.

**Review by aluarius — APPROVED** [review]
Not sure if it related to this pr, but if you're using light theme of the nuxt-docs and then go to http://localhost:3000/en/styles/colors, it automatically switches to the dark one.

That pretty weird.

**Pull request merged**